### PR TITLE
Build MkDocs Lua API docs using GitHub CI, deploy to api.minetest.net

### LINKED ISSUE #13072: API docs: Build using GitHub and host under api.minetest.net
Currently lua_api.txt gets generated into a pretty HTML documentation hosted by Gitlab pages, available at https://minetest.gitlab.io/minetest/. However, I think it would be better if it existed under api.minetest.net, it'd be shorter, easier to remember and more intuitive (ah, it's the API documentation!) compared to previously.  [Gitlab pages does allow you to host sites under a custom domain](https://docs.gitlab.com/ee/user/project/pages/custom_domains_ssl_tls_certification/), just like Github.

### PR DESCRIPTION
**Pull request by rollerozxa**
Closes #13072. 

This PR adds a workflow script to build the MkDocs Lua API using GitHub CI and deploying it using GitHub pages. As per #13072, it gets deployed to the shorter and neater api.minetest.net subdomain.

Also removes a patch to the codehilite source to always set the code highlight language to Lua, this isn't necessary as of #13439. (the patch fails to apply now with a newer version anyways)

## Requirements before merge:
1. There needs to be a DNS CNAME record for `api` added pointing to `minetest.github.io`, under the `minetest.net` domain.
2. The GitHub pages build source needs to be set to "GitHub Actions" in the `minetest/minetest` repository:

image

## To do
This PR is Ready for Review.

## How to test
I tested the workflow at https://github.com/KetchupLand/minetest_hhh, which got deployed at https://ketchupland.voxelmanip.se/minetest_hhh/. So the workflow script works.

### CODE REVIEW

File: .github/workflows/lua_api_deploy.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        uses: actions/configure-pages@v3
+
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v2
+        with:
+          path: 'public/'
+
+      - name: Deploy to GitHub Pages
+        id: deployment
+        uses: actions/deploy-pages@v2
```
Comment (rubenwardy) [inline .github/workflows/lua_api_deploy.yml:48]:
```suggestion
        uses: actions/deploy-pages@v2

```

File: CNAME [hunk 2]
```diff
@@ -0,0 +1 @@
+api.minetest.net
```
Comment (rubenwardy) [inline CNAME:1]:
```suggestion
api.minetest.net

```

File: .github/workflows/lua_api_deploy.yml (hunk 1 above)
Comment (rubenwardy) [inline .github/workflows/lua_api_deploy.yml:48]:
```suggestion
        uses: actions/deploy-pages@v2

```

**Review by rubenwardy — APPROVED** [review]
Looks good to me, please just fix your line endings

**Comment by sfan5** [comment]
Are the "Requirements before merge:" met?

**Comment by srifqi** [comment]
The `CNAME` DNS record has not been set (checked using Cloudflare servers:

```
C:\>nslookup -q=cname api.minetest.net
Server:  one.one.one.one
Address:  1.0.0.1

*** one.one.one.one can't find api.minetest.net: Non-existent domain
```

... and also using [Google Public DNS](https://dns.google/query?name=api.minetest.net&rr_type=CNAME)).

**Comment by rollerozxa** [comment]
Yeah this is still waiting on @celeron55 to set up the DNS and repository settings.

**Comment by celeron55** [comment]
I made the DNS and Github configs now, after forgetting about this PR too many times.

I see that after this, there will be two different sets of pages on minetest.github.io, one being generated from this main repo and the other from https://github.com/minetest/minetest.github.io, and they are differentiated by the CNAME files in each repo.

**Pull request merged**